test_apps/smokeTests/gridTools.cpp

```diff
@@ -0,0 +1,620 @@
+                grid->SetValueIJK( i, j, k, value );
+                value += xIncrement;
+            }
+            value += yIncrement;
+        }
+        value += zIncrement;
+    }
+}
+
+// This function iterates across all nodes in a grid; making comparisons between the
+// data values returned by the functions GetValueAtIndex() and GetValue().
+bool CompareIndexToCoords( 
+    VAPoR::Grid* grid, 
+    double &rms,                // Root Mean Square error
+    size_t &numMissingValues,   // Counter for receiving MissingValue upon query
+    size_t &disagreements      // Counter for when AccessIJK() and GetValue() disagree
+) {
+    bool rc = true;
+
+    rms   = 0.f;
+    disagreements  = 0;
+    numMissingValues = 0;
+
+    std::vector< size_t > dims = grid->GetDimensions();
+    size_t x = dims[X];
+    size_t y = dims[Y];
+    size_t z = 1;
+    if ( dims.size() == 3 ) 
+        z = dims[Z];
+ 
+    double peak  = 0.f;
+    double sum     = 0;
+    for ( size_t k=0; k<z; k++ ) {
+        for ( size_t j=0; j<y; j++ ) {
+            for ( size_t i=0; i<x; i++ ) {
+                size_t indices[3] = { i, j, k };
+                //float trueValue = grid.AccessIJK( i, j, k );
+                double trueValue = grid->GetValueAtIndex( {i,j,k} );
+                
+                double coords[3];
+                grid->GetUserCoordinates( indices, coords );
+                float sampleValue = grid->GetValue( coords );
+                
+                if (sampleValue == grid->GetMissingValue()) {
+                    numMissingValues++;
+                    continue;
+                }
+
+                double error = abs( sampleValue - trueValue );
+
+                if (error != 0) {
+                    disagreements++;
+                }
+
+                if ( error > peak ) 
+                    peak = error;
+                sum += error*error;
+
+            }
+        }
+    }
+
+    rms = sqrt( sum / (x*y*z) );
+
+    if ( rms != 0 || disagreements > 0 )
+        rc = false;
+    return rc;
+}
+
+bool isNotEqual(
+    double x, 
+    double y
+) {
+  const double epsilon = 1e-5;
+  return std::abs(x - y) > epsilon * std::abs(x);
+}
+
+bool TestConstNodeIterator( 
+    const Grid *g, 
+    size_t& count, 
+    size_t& expectedCount,
+    size_t& disagreements, 
+    double& time 
+) {
+    bool rc = true;
+    count = 0;
+    expectedCount = 1;
+    disagreements = 0;
+
+    Grid::ConstNodeIterator itr;
+    Grid::ConstNodeIterator enditr = g->ConstNodeEnd();
+
+    double t0 = Wasp::GetTime();
+    itr = g->ConstNodeBegin();
+    
+    std::vector< size_t > dims = g->GetDimensions();
+    for ( auto dim : dims )
+        expectedCount *= dim;
+
+    for ( ; itr!=enditr; ++itr) {
+        std::vector<size_t> ijk = Wasp::VectorizeCoords( count, dims );
+       
+        double itrData  = g->GetValueAtIndex( (*itr).data() );
+        double gridData = g->AccessIJK( ijk[X], ijk[Y], ijk[Z] ); 
```

Comment:
This will result in a memory access violation if ijk does not have at least three elements. Suggest you use GetValueAtIndex instead of AccessIJK. The former you can simply pass ijk directly (it accepts a 'const std::vector <size_t> &' argument). The latter should be deprecated, and simply calls the former.

Reply:
done